File: 백준/Bronze/25304. 영수증/영수증.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
	      Scanner in = new Scanner(System.in);
	      int m = in.nextInt();
	      int n = in.nextInt();
	      for(int i = 0; i < n; i++){
	          m = m -(in.nextInt()*in.nextInt());
	      }
	      if(m != 0)
	          System.out.print("No");
	      else
	          System.out.print("Yes");
    }
}

File: 백준/Bronze/25304. 영수증/영수증.kt
import java.util.Scanner
fun main(){
    val sc = Scanner(System.`in`)
    var m = sc.nextInt()
    var n = sc.nextInt()
    for(i in 1..(n))
    m -= (sc.nextInt() * sc.nextInt())
    if(m == 0)
    print("Yes")
    else
    print("No")
}